src/main/java/codeu/controller/ActivityfeedServlet.java

```diff
@@ -68,13 +68,25 @@ public void setMessageStore(MessageStore messageStore) {
 		for (Conversation convo : conversations) {
 			User owner = userStore.getUser(convo.getOwnerId());
 			convo.setDisplayText(owner.getName() + " created conversation: " + convo.getTitle());
-			activity.add(convo);
+			// Only show activity about this convo if it is public or the logged-in user is a member
+			if (convo.getIsPublic()) {
+				activity.add(convo);
+			} else if ((username != null) && 
+					userStore.getUser(username).getConversations().contains(convo.getId())) {
+				activity.add(convo);
+			}
```

Comment:
We could use this same concept in the conversation page when displaying public & group conversations